### PR TITLE
Goroutines

### PR DESCRIPTION
**Pull request by iamargus95**
Could not get the flags to work. Printing from channel also fails. Logic is correct in gogrep.go, Needs refactoring in the main.go

### CODE REVIEW

**Comment by iamargus95** [comment]
The program works as expected. Any suggestions on a way to improve concurrency would be a great help.

File: cmd/gogrep/main.go [hunk 1]
```diff
@@ -21,44 +23,61 @@ func main() {
...
+	}
 
-		} else if count {
-			output := gogrep.GrepCount(fileContents, pattern)
-			fmt.Println(output)
+	outputChan := make(chan []string)
+	var wg sync.WaitGroup
+	for _, path := range paths {
+		wg.Add(1)
+		go worker(caseSensitive, count, after, before, path, pattern, outputChan, &wg)
```
Comment (jdk2588) [inline cmd/gogrep/main.go]: Why are we checking all the rules inside a goroutine? It is a waste of computation.
If you already know which flag will be used, why not do it outside?

File: cmd/gogrep/main.go [hunk 2]
```diff
@@ -21,44 +23,61 @@ func main() {
...
+	}
 
-		} else if after > 0 {
-			output := gogrep.GrepAfter(after, fileContents, pattern)
+	go func() {
 
+		for output := range outputChan {
 			for i := 0; i < len(output); i++ {
 				fmt.Println(output[i])
```
Comment (jdk2588) [inline cmd/gogrep/main.go]: This is an impure function, I cannot test it. Better to push into a data structure and use outside of goroutines.

File: iofile/listFiles.go [hunk 3]
```diff
@@ -1,25 +1,46 @@
 package iofile
```
Comment (jdk2588) [inline iofile/listFiles.go]: `iofile` means what?

**Review by jdk2588 — CHANGES_REQUESTED** [review]

File: iofile/listFiles.go (hunk 3 above)
Comment (iamargus95) [reply iofile/listFiles.go]: I used iofile as input/output file as I was performing all file related activities in this package.

I could change it to something more in line with the expected nomenclature.

File: cmd/gogrep/main.go (hunk 2 above)
Comment (iamargus95) [reply cmd/gogrep/main.go]: Okay, Will create a separate function to WriteToStdout.

File: cmd/gogrep/main.go (hunk 1 above)
Comment (iamargus95) [reply cmd/gogrep/main.go]: Do I create workers for every flag and invoke them by checking the flags within main?

File: iofile/listFiles.go (hunk 3 above)
Comment (jdk2588) [reply iofile/listFiles.go]: Standard practise is to call as `io`.

File: cmd/gogrep/main.go (hunk 1 above)
Comment (jdk2588) [reply cmd/gogrep/main.go]: Is it possible to use multiple flags at a time?

Comment (iamargus95) [reply cmd/gogrep/main.go]: No, Not yet. But I am using both boolVar and Intvar flags, So I need to pass the variables from the Command line to the function as int. I'm almost done with it.

Comment (iamargus95) [reply cmd/gogrep/main.go]: Created separate worker functions for each flag which get executed after checking the condition  outside goroutines.

**Pull request merged**